### PR TITLE
feat: improve generated comments

### LINKED ISSUE #2580: Improve generated comments with user defined .Comment("")
- [x] I have searched the [issues](https://github.com/ent/ent/issues) of this repository and believe that this is not a duplicate.

## Summary 💡

As discussed in #944, the `.Comment("")` method can be used to add a custom comment to the fields of a generated entity. However, there are two significant limitations to this feature that I've noticed so far. 

Firstly, the custom comment is inserted only after the default comment, which looks like `CreatedAt holds the value of the "created_at" field.`. Aside from having a comment for comment's sake, this default comment provides essentially no value to me when developing. It tells me nothing I don't already know by typing the name of the field, and instead can be distracting. My proposal is that if the developer defines a comment with the `.Comment()` method, this comment should _replace_ the default one.

Secondly, while the `.Comment()` method can be used on edges, those comments are not currently included in the generated entity code. Instead the default comment is once again used (`Parent holds the value of the parent edge.`). This is confusing because the "value of the edge" seems more akin to the weight or other value assigned to the edge itself, as opposed to the entity(s) on the other end of it. My proposal is that once again if the developer defines a comment with the `.Comment()` method, this comment should replace the default one.

If these proposals are acceptable I would be happy to start a pull request!

## Motivation 🔦

Writing code that is well documented is one of the most important ports of writing good code in my opinion, and generated code is no exception. Especially with modern IDEs which provide instant access to comments and documentation when typing, having the information that they contain be relevant is important. 

Consider the following two examples:
- With current implementation:
  image

- After desired implementation (user can specify edge comment):
  image

Other references:
- [Godoc: documenting Go code (go.dev blog)](https://go.dev/blog/godoc)
- [The Value of Code Documentation](https://www.olioapps.com/blog/the-value-of-code-documentation/)

### PR DESCRIPTION
**Pull request by willbicks**
This feature improves the comments generated for entities by improving the impact of the `.Comment("")` method on fields and edges. The details and justification are described in #2580.

The following changes are implemented:

- If a comment is specified on a field schema with the `.Comment("")` method, that comment now _replaces_ the default comment instead of being appended to it.
- If a comment is specified on an edge schema with the `.Comment("")` method, that comment replaces the default comment instead of being ignored.
- Documentation of the `.Comment("")` method is added to the getting started guide, and the field and edge schema docs.

Closes #2580

### CODE REVIEW

**Comment by willbicks** [comment]
@a8m just wanted to flag this for your review in case you hadn't seen it, would love to know if you have any input!

**Comment by a8m** [comment]
Thanks for the contribution @willbicks 🙏

I'll review this later today. The CI is red - you need to run `go generate ./...` from the root of the project.

**Comment by willbicks** [comment]
@a8m Thanks for the tip, I updated the relevant integration schemas to reflect this change and regenerated all the generated files.

File: doc/md/getting-started.md [hunk 1]
```diff
@@ -84,9 +84,11 @@ import (
 func (User) Fields() []ent.Field {
 	return []ent.Field{
 		field.Int("age").
-			Positive(),
+			Positive().
+			Comment("Age of the user in years."),
 		field.String("name").
-			Default("unknown"),
+			Default("unknown").
+			Comment("Name (first last) of the user. Defaults to \"unkown\"."),
```
Comment (a8m) [inline doc/md/getting-started.md:91]: You can add it to the fields/edges pages, no need for this in the getting-started page.

File: doc/md/getting-started.md [hunk 2]
```diff
@@ -310,7 +312,8 @@ Let's add the `"cars"` edge to the `User` schema, and run `go generate ./ent`:
 // Edges of the User.
 func (User) Edges() []ent.Edge {
 	return []ent.Edge{
-		edge.To("cars", Car.Type),
+		edge.To("cars", Car.Type).
+			Comment("Cars owned by this user."),
```
Comment (a8m) [inline doc/md/getting-started.md:316]: gitto

File: doc/md/schema-edges.mdx [hunk 3]
```diff
@@ -1416,6 +1416,23 @@ However, you should note, that this is currently an SQL-only feature.
...
+A comment can be added to the edge using the `.Comment()` method. This comment
+appears before the edge in the generated entity code. Newlines are supported 
+using the `\n` escape sequence.
+
+```go
+// Edges of the User.
+func (User) Edges() []ent.Edge {
+	return []ent.Edge{
+		edge.To("pets", Pet.Type).
+			Comment("Pets that this user is responsible for taking care of. \n
```
Comment (a8m) [inline doc/md/schema-edges.mdx:1430]:
```suggestion
			Comment("Pets that this user is responsible for taking care of.\n 
```

File: doc/md/schema-fields.md [hunk 4]
```diff
@@ -508,6 +508,23 @@ func (User) Fields() []ent.Field {
...
+appears before the field in the generated entity code. Newlines are supported 
+using the `\n` escape sequence.
+
+```go
+// Fields of the user.
+func (User) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("name").
+			Default("John Doe").
+			Comment("Name of the user. \n If not speicifed, defaults to \"John Doe\"."),
```
Comment (a8m) [inline doc/md/schema-fields.md:523]:
```suggestion
			Comment("Name of the user.\n If not specified, defaults to \"John Doe\"."),
```

File: entc/gen/template/ent.tmpl [hunk 5]
```diff
@@ -207,11 +207,23 @@ func ({{ $receiver }} {{ $slice }}) config(cfg config) {
...
 {{/* A template for setting the field comment. */}}
 {{- define "model/fieldcomment" }}
-	// {{ $.StructField }} holds the value of the "{{ $.Name }}" field.
 	{{- if $.Comment }}
 		{{- range $line := split $.Comment "\n" }}
 			// {{ $line }}
 		{{- end }}
+    {{- else }}
+        // {{ $.StructField }} holds the value of the "{{ $.Name }}" field.
+	{{- end }}
```
Comment (a8m) [inline entc/gen/template/ent.tmpl:216]:
```suggestion
	{{- if $.Comment }}
		{{- range $line := split $.Comment "\n" }}
			// {{ $line }}
		{{- end }}
	{{- else }}
		// {{ $.StructField }} holds the value of the "{{ $.Name }}" field.
	{{- end }}
```

File: entc/gen/template/ent.tmpl [hunk 6]
```diff
@@ -207,11 +207,23 @@ func ({{ $receiver }} {{ $slice }}) config(cfg config) {
...
+
+{{/* A template for setting the edge comment. */}}
+{{- define "model/edgecomment" }}
+	{{- if $.Comment }}
+		{{- range $line := split $.Comment "\n" }}
+			// {{ $line }}
+		{{- end }}
+    {{- else }}
+        // {{ $.StructField }} holds the value of the {{ $.Name }} edge.
 	{{- end }}
```
Comment (a8m) [inline entc/gen/template/ent.tmpl:227]:
```suggestion
	{{- else }}
		// {{ $.StructField }} holds the value of the {{ $.Name }} edge.
	{{- end }}
```

File: entc/integration/edgeschema/ent/schema/tweet.go [hunk 7]
```diff
@@ -31,6 +31,6 @@ func (Tweet) Edges() []ent.Edge {
 		edge.From("user", User.Type).
 			Ref("tweets").
 			Through("tweet_user", UserTweet.Type).
-			Comment("The uniqueness is enforced on the edge schema"),
+			Comment("User who authored this tweet. \nThe uniqueness is enforced on the edge schema"),
```
Comment (a8m) [inline entc/integration/edgeschema/ent/schema/tweet.go:34]: No need to change this comment, please. The second sentence is more important for this example.

**Review by a8m — APPROVED** [review]
Amazing job!

I added a few small comments. Please address them and I'll merge it. Thanks <3

**Comment by willbicks** [comment]
@a8m, I was able to push a new version that addressed your suggestions.

I have a few more places where I think the comment generation functionality could be improved, so I'll be looking into those and submit some requests when I have something.

Thanks for all your work in maintaining this project!

**Review by a8m — APPROVED** [review]
Excellent job 🚀

**Pull request merged**